DllImportGenerator/DllImportGenerator/StubCodeContext.cs

```diff
@@ -0,0 +1,314 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.Interop
+{
+    class StubCodeContext
```

Comment:
One thing we'll likely want to do in the future is refactor this into a base class that has the information that the marshalling generators will need and a subclass that has the methods that actually build up the method signature and calls the marshallers. That way we can share the base class for native->managed code-gen as well as struct marshalling (which would each have separate subclasses that build up the surrounding code infrastructure for their particular marshalling needs).

Reply:
Split into `StubCodeContext` base and `StubCodeGenerator` subclass now, and we can modify/refactor as needed in the future.